src/components/LocationSelector/LocationSelectorWithGoogleLoader.js

```diff
@@ -0,0 +1,242 @@
+        })
+    ).isRequired,
+    /** default radius value */
+    radiusDefaultValue: PropTypes.number,
+    /** radius measurement unit */
+    radiusUnits: PropTypes.oneOf(['km', 'mi']).isRequired,
+    /** radius label renderer e.g. radius => `+ ${radius} km` */
+    renderRadiusLabel: PropTypes.func.isRequired,
+    /** min radius value of the slider component */
+    minRadius: PropTypes.number,
+    /** max radius value of the slider component */
+    maxRadius: PropTypes.number,
+    /** radius step value of the slider component */
+    radiusStep: PropTypes.number,
+    /** country where search can take place */
+    country: PropTypes.string.isRequired,
+    /**
+     * type of locations that should be searched for.
+     * For details see: https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompletionRequest.types
+     */
+    placeTypes: PropTypes.arrayOf(PropTypes.string),
+    /** show country name in autocomplete suggestions */
+    showCountryInSuggestions: PropTypes.bool,
+    /** A title for the modal that will be used by screenreaders */
+    modalContentLabel: PropTypes.string.isRequired,
+    /** placeholder for both main field and autocomplete field in modal */
+    inputPlaceholder: PropTypes.string.isRequired,
+    /** placeholder for empty LocationAutocomplete list */
+    noSuggestionsPlaceholder: PropTypes.string.isRequired,
+    /** function to be executed if error occurs while fetching suggestions */
+    onLocationAutocompleteError: PropTypes.func,
+    /** string to be displayed in FieldWrapper when the modal is closed, but locations are selected */
+    selectionPlaceholder: PropTypes.string,
+    /** label for the Done button */
+    doneLabel: PropTypes.string.isRequired,
+    /** label to be used on the clear all button */
+    clearLabel: PropTypes.string.isRequired,
+    /** function called with location object as an argument when it is selected from the suggestions */
+    onAddLocation: PropTypes.func.isRequired,
+    /** function called with a location details as an argument to be changed */
+    onUpdateLocation: PropTypes.func.isRequired,
+    /** function with a locationId as an argument to be removed */
+    onRemoveLocation: PropTypes.func.isRequired,
+    /** callback function for the Clear button click */
+    onRemoveAllLocations: PropTypes.func.isRequired,
+    /** callback function for closed modal */
+    onBlur: PropTypes.func,
+};
+
+LocationSelectorWithGoogleLoader.defaultProps = {
+    radiusDefaultValue: 1,
+    minRadius: 1,
+    maxRadius: 100,
+    radiusStep: 1,
+    showCountryInSuggestions: true,
+    onBlur: () => null,
+    additionalGoogleProps: {},
+    region: undefined,
+    selectionPlaceholder: null,
+    placeTypes: LocationAutocomplete.defaultProps.placeTypes,
+    onLocationAutocompleteError: LocationAutocomplete.defaultProps.onError,
+};
+
+export default LocationSelectorWithGoogleLoader;
```

Comment:
`LocationSelector` and `LocationSelectorWithGoogleLoader` are mixed together now in the same directory.
I think they could be better separated into two different components living in different folders:
`LocationSelector` is kind of a `Molecule`
`LocationSelectorWithGoogleLoader` is an `Organism`.

Also, as Oleksii @mukiienko noticet, both `LocationSelector` and `LocationSelectorWithGoogleLoader` are referencing the same `scss` which shouldn't be the case.